RadarSDK/RadarKVStore.m

```diff
@@ -0,0 +1,441 @@
+}
+
+- (double)doubleForKey:(NSString *)key {
+    NSData *data = [self.fileHandler readFileAtPath: [self getSettingFilePath:key]];
+    if (!data) {
+        return 0;
+    }
+    double value;
+    [data getBytes:&value length:sizeof(value)];
+    return value;
+}
+
+- (void)setDouble:(double)value forKey:(NSString *)key {
+    [self.fileHandler writeData:[NSData dataWithBytes:&value length:sizeof(value)] toFileAtPath:[self getSettingFilePath:key]];
+}
+
+- (NSObject *_Nullable)objectForKey:(NSString *)key {
+    NSData *data = [self.fileHandler readFileAtPath: [self getSettingFilePath:key]];
+    if (!data) {
+        return nil;
+    }
+    NSObject *value = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+    return value;
+}
+
+- (void)setObject:(NSObject *_Nullable)value forKey:(NSString *)key {
+    if (!value) {
+        [self.fileHandler deleteFileAtPath:[self getSettingFilePath:key]];
+        return;
+    }
+    [self.fileHandler writeData:[NSKeyedArchiver archivedDataWithRootObject:value] toFileAtPath:[self getSettingFilePath:key]];
+}
+
+- (void)removeObjectForKey:(NSString *)key {
+    [self.fileHandler deleteFileAtPath:[self getSettingFilePath:key]];
+}
+
+- (NSInteger)integerForKey:(NSString *)key {
+    NSData *data = [self.fileHandler readFileAtPath: [self getSettingFilePath:key]];
+    if (!data) {
+        return 0;
+    }
+    NSInteger value;
+    [data getBytes:&value length:sizeof(value)];
+    return value;
+}
+
+- (void)setInteger:(NSInteger)value forKey:(NSString *)key {
+    [self.fileHandler writeData:[NSData dataWithBytes:&value length:sizeof(value)] toFileAtPath:[self getSettingFilePath:key]];
+}
+
+- (void)removeAllObjects {
+    NSArray<NSString *> *allSettings = [self.fileHandler sortedFilesInDirectory:self.settingsFileDir];
+    for (NSString *setting in allSettings) {
+        [self.fileHandler deleteFileAtPath:[self getSettingFilePath:setting]];
+    }
+    [[NSFileManager defaultManager] createDirectoryAtPath:self.settingsFileDir withIntermediateDirectories:YES attributes:nil error:nil];
+}
+
+- (BOOL)keyExists:(NSString *)key {
+    return [[NSFileManager defaultManager] fileExistsAtPath:[self getSettingFilePath:key]];
+}
+
+- (NSString *)doubleWriteStringGetter:(NSString *)key {
+    NSString *radarKVStoreRes = [self stringForKey:key];
+    if ([RadarSettings useRadarKVStore]) {
+        return radarKVStoreRes;
+    }
+    NSString *nsUserDefaultsRes = [[NSUserDefaults standardUserDefaults] stringForKey:key];
+    if ((radarKVStoreRes && ![radarKVStoreRes isEqualToString:nsUserDefaultsRes]) || (nsUserDefaultsRes && ![nsUserDefaultsRes isEqualToString:radarKVStoreRes])) {
+        [[RadarLogBuffer sharedInstance] write:RadarLogLevelError type:RadarLogTypeSDKError message:[NSString stringWithFormat:@"Discrepencey with NSUserDefault %@ mismatch.", key]];
+    }
+    return nsUserDefaultsRes;
+}
+
+- (void)doubleWriteStringSetter:(NSString *)key value:(NSString *)value {
```

Comment:
These names are painful.... can we just do `getString`, `setString` and so forth...?